From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Mon, 17 Nov 2008 15:08:21 +0000 (-0600)
Subject: rtl8187: Fix transmission count sent to mac80211
X-Git-Tag: v2.6.29-rc1~581^2~383^2~43
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fstable%2Flinux-2.6.29.y.git;a=commitdiff_plain;h=1548c86ab1552199a3b99b958ccfb590fc9c8f84

rtl8187: Fix transmission count sent to mac80211

In the commit entitled "mac80211/drivers: rewrite the rate control
API", the meaning of the packet transmit count was changed from the
number of retries to the total number.  In driver rtl8187, this change
was missed.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
Acked-by: Herton Ronaldo Krzesinski <herton@mandriva.com.br>
Tested-by: Hin-Tak Leung <htl10@users.sourceforge.net>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
---

diff --git a/drivers/net/wireless/rtl818x/rtl8187_dev.c b/drivers/net/wireless/rtl818x/rtl8187_dev.c
index 180bc50..69ea522 100644
--- a/drivers/net/wireless/rtl818x/rtl8187_dev.c
+++ b/drivers/net/wireless/rtl818x/rtl8187_dev.c
@@ -496,7 +496,7 @@ static void rtl8187b_status_cb(struct urb *urb)
 			__skb_unlink(skb, &priv->b_tx_status.queue);
 			if (tok)
 				info->flags |= IEEE80211_TX_STAT_ACK;
-			info->status.rates[0].count = pkt_rc;
+			info->status.rates[0].count = pkt_rc + 1;
 
 			ieee80211_tx_status_irqsafe(hw, skb);
 		}
